sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GlobalAddressResolver.java

```diff
@@ -77,46 +80,87 @@ public GlobalAddressResolver(
         }
     }
 
+    public void dispose() {
+        for (EndpointCache endpointCache : this.addressCacheByEndpoint.values()) {
+            endpointCache.addressCache.dispose();
+        }
+    }
+
+    @Override
+    public URI getAddressResolverURI(RxDocumentServiceRequest rxDocumentServiceRequest) {
+        return this.endpointManager.resolveServiceEndpoint(rxDocumentServiceRequest);
+    }
+
+    @Override
+    public boolean removeAddressResolverURI(RxDocumentServiceRequest request) {
+        final URI uri = this.getAddressResolverURI(request);
+        this.addressCacheByEndpoint.computeIfPresent(uri, (address, cache) -> {
+            // TODO (DANOBLE) What work must be done here?
```

Comment:
as discussed on the code review session. here you need to do a surgery in GatewayAddressCache to remove the corresponding stale replica address.

Reply:
Please fill the implementation gap.